python/tank/platform/bundle.py

```diff
@@ -823,9 +821,28 @@ def __resolve_setting_value(self, key, value):
             schema = self.__descriptor.get_configuration_schema().get(key)
         except:
             schema = None
-        
+
         if schema:
-            # post process against schema
-            value = self.__post_process_settings_r(key, value, schema)
+
+            # If the value is None, then the setting wasn't specified. Use the
+            # default value in the schema instead.
+            if value is None:
```

Comment:
What happens if None is a valid value?  Is there a way to know if the setting is actually missing?